Support Chat: Fix responsiveness and z-index issues

jasmussen opened the pull request:
This PR fixes two issues:

1. Panel responsiveness
2. Z-index issues.

To test 1:

- Click support chat link in bottom of the sidebar
- Make the window small so the new right-most sidebar pane becomes a small panel
- Verify that the panel has a drop shadow and a consistent sensible size

To test 2:

- Open the panel like above
- Visit My Sites > Stats > Insights
- Make the browser window short, so the panel overlaps the posting activity. The posting activity has a gradient that would previously overlap the panel, it shouldn't anymore.

Screenshots:

![screen shot 2016-09-29 at 11 59 25](https://cloud.githubusercontent.com/assets/1204802/18949581/4d48c1e4-863c-11e6-9391-d04b6e92e20d.png)


![screen shot 2016-09-29 at 11 55 44](https://cloud.githubusercontent.com/assets/1204802/18949578/4a9fdeaa-863c-11e6-8150-744d93c36a97.png)

*Note*: I read the [CSS guidelines](https://github.com/Automattic/wp-calypso/blob/master/docs/coding-guidelines/css.md#z-index) on using `$z-index` to fix issue 2, but wasn't able to get this working. If that's a requirement, I would appreciate help on getting this working. I kept getting various versions of this error message:

```
WARNING: No layer found for `.happychat .happychat__container` of `['.happychat" ".happychat__container']` in $z-layers map. Property omitted.
```

gwwar commented on client/components/happychat/style.scss:
```diff
@@ -63,7 +63,7 @@
 		background: lighten( $gray, 30% );
 		bottom: 0;
-		z-index: 10;
+		z-index: 1020;
```
Why 1020? This should also be in our z-index map. The warning you were seeing earlier meant that that value was never used in any css file.

gwwar replied:
<img width="972" alt="screen shot 2016-09-29 at 9 02 26 am" src="https://cloud.githubusercontent.com/assets/1270189/18961917/84a1927e-8623-11e6-9dd7-82fa84f612d0.png">

The rule becomes:
```scss
z-index: z-index( 'root' '.happychat__container.is-open' );
```

And in _z-index.scss we add:
```
'root': (
//...
'.happychat__container.is-open': 1020 //You may want to change this value depending on what you want to be on top of. 
```

gwwar commented:
@jasmussen try inspecting the element with this extension https://github.com/gwwar/z-context You should be able to find the parent context more easily there.

jasmussen commented:
Hey @gwwar, thanks a ton for your review and generous z-index help. Makes much more sense now.

I rebased and changed the code to use your mixin now, much nicer. No, there wasn't any specific reason for using 1020 as the z-index, other than it has to be above pretty much anything. Edited that to 999 now. Looks good:

![screen shot 2016-09-30 at 13 17 47](https://cloud.githubusercontent.com/assets/1204802/18990215/bd78eae6-8710-11e6-8846-9cba6693f5b0.png)

gwwar commented:
👍 Looks like this behaves pretty well for desktop. I think we'll probably want some refinements on smaller viewports ( the chat box takes up a weird amount of space, maybe snapping across the bottom is better ) but that can be handled in follow up PRs

jasmussen commented:
Thanks, I'll merge this when I get back to my desktop.

> I think we'll probably want some refinements on smaller viewports

It's not perfect indeed, but the system detects if mobile and redirects to a dedicated page. We had to do this because performance with just a responsive design wasn't very good, and there was breaking scroll bleed when the responsive chat box covered the reader.
